### PR TITLE
Use == and != instead of Equals(typeof(..))

### LINKED ISSUE #621: Replace `Equals(typeof(...))` with more efficient equvalent
CoreLib and Reflection for NativeAOT uses `object.Equals` to check for type equality (e.g. `type.Equals(typeof(string))`) in number of places. `==` or `object.ReferenceEquals` are more efficient ways to do this when we are dealing with runtime types.

Full context https://github.com/dotnet/runtimelab/pull/616#discussion_r567575204

### PR DESCRIPTION
**Pull request by kant2002**
This more efficient way for runtime types
Closes #621

### CODE REVIEW

**Comment by jkotas** [comment]
Could you please take a look at the build breaks?

File: src/coreclr/nativeaot/System.Private.Reflection.Core/src/System/Reflection/Runtime/TypeInfos/RuntimeTypeInfo.cs [hunk 1]
```diff
@@ -880,11 +880,11 @@ private TypeClassification Classification
...
                         Type valueType = typeof(ValueType);
 
-                        if (baseType.Equals(enumType))
+                        if (baseType == enumType)
                             classification |= TypeClassification.IsEnum | TypeClassification.IsValueType;
-                        if (baseType.Equals(typeof(MulticastDelegate)))
+                        if (baseType == typeof(MulticastDelegate))
                             classification |= TypeClassification.IsDelegate;
-                        if (baseType.Equals(valueType) && !(this.Equals(enumType)))
+                        if (baseType == valueType && this != enumType))
```
Comment (MichalPetryka) [inline src/coreclr/nativeaot/System.Private.Reflection.Core/src/System/Reflection/Runtime/TypeInfos/RuntimeTypeInfo.cs:887]:
```suggestion
                        if (baseType == valueType && this != enumType)
```

File: src/coreclr/nativeaot/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/ExecutionEnvironmentImplementation.MappingTables.cs [hunk 2]
```diff
@@ -501,7 +501,7 @@ private RuntimeTypeHandle[] GetDynamicInvokeInstantiationArguments(MethodBase re
             Type returnType = reflectionMethodInfo != null ? reflectionMethodInfo.ReturnType : typeof(void);
 
             // Only use the return type if it's not void
-            if (!returnType.Equals(typeof(void)))
+            if (!returnType == typeof(void))
```
Comment (MichalPetryka) [inline src/coreclr/nativeaot/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/ExecutionEnvironmentImplementation.MappingTables.cs:504]:
```suggestion
            if (returnType != typeof(void))
```

**Comment by kant2002** [comment]
Sorry. Was too optimistic

**Review by jkotas — APPROVED** [review]
Thanks!

**Pull request merged**